src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -939,3 +957,222 @@ void AppHost::_SummonWindowRequested(const winrt::Windows::Foundation::IInspecta
     summonArgs.ToggleVisibility(false); // Do not toggle, just make visible.
     _HandleSummon(sender, summonArgs);
 }
+
+void AppHost::_MinimizeToTrayRequested(const winrt::Windows::Foundation::IInspectable&,
+                                       const winrt::Windows::Foundation::IInspectable&)
+{
+    _window->HideWindow();
+}
+
+void AppHost::_HandleTrayIconPressed()
+{
+    // No name in the args means summon the mru window.
+    Remoting::SummonWindowSelectionArgs args{};
+    args.SummonBehavior().ToggleVisibility(false);
+    _windowManager.SummonWindow(args);
+}
+
+// Method Description:
+// - Creates and adds an icon to the notification tray.
+// If an icon already exists, update the HWND associated
+// to the icon with this window's HWND.
+// Arguments:
+// - <unused>
+// Return Value:
+// - <none>
+void AppHost::_UpdateTrayIcon()
+{
+    if (_logic.GetAlwaysShowTrayIcon() || _logic.GetMinimizeToTray())
+    {
+        if (!_trayIconData)
+        {
+            NOTIFYICONDATA nid{};
+            nid.cbSize = sizeof(NOTIFYICONDATA);
+
+            // This HWND will receive the callbacks sent by the tray icon.
+            nid.hWnd = _window->GetHandle();
+
+            // App-defined identifier of the icon. The HWND and ID are used
+            // to identify which icon to operate on when calling Shell_NotifyIcon.
+            // Multiple icons can be associated with one HWND, but here we're only
+            // going to be showing one so the ID doesn't really matter.
+            nid.uID = 1;
+
+            nid.uCallbackMessage = CM_NOTIFY_FROM_TRAY;
+
+            nid.hIcon = static_cast<HICON>(GetActiveAppIconHandle(true));
+            StringCchCopy(nid.szTip, ARRAYSIZE(nid.szTip), RS_(L"AppName").c_str());

```

Comment:
Do we need to distinguish between "Windows Terminal", "Windows Terminal Preview", and "Windows Terminal (Dev)" here?

Reply:
For now I'll be keeping it as just "Windows Terminal" in all cases, as apparently differentiating between the three of them is kind of annoying and hard.